cr-examples/triton/src/main/java/oracle/code/triton/TensorType.java

```diff
@@ -73,16 +75,17 @@ public int hashCode() {
     }
 
     @Override
-    public String toString() {
-        StringBuilder s = new StringBuilder();
-
-        s.append(NAME);
-        s.append("<");
-        s.append(shape.stream().map(i -> "x" + i).collect(Collectors.joining(",")));
-        s.append(",");
-        s.append(fromType(eType));
-        s.append(">");
+    public TypeDefinition toTypeDefinition() {
+        List<TypeDefinition> args = new ArrayList<>();
+        for (int i : shape) {
+            args.add(new TypeDefinition("x" + i, List.of()));
+        }
+        args.add(fromType(eType).toTypeDefinition());
+        return new TypeDefinition(NAME, args);
+    }
 
-        return s.toString();
+    @Override
+    public String toString() {
```

Comment:
Would it make sense for this to be a default method in the TypeElement interface?

Reply:
Yes, good point, i will do that then integrate.

Reply:
Ah i forgot, it's not possible for default methods to override the methods on Object.